controllers/addon/monitoring_federation_reconciler_test.go

```diff
@@ -57,32 +64,65 @@ func TestEnsureMonitoringFederation_MonitoringPresentInSpec_NotPresentInCluster(
 			assert.Equal(t, GetMonitoringNamespaceName(addon), namespace.Name)
 		}).
 		Return(nil)
-	c.On("Get", testutil.IsContext, mock.IsType(types.NamespacedName{}), mock.IsType(&monitoringv1.ServiceMonitor{}), mock.Anything).
+
+	secretADOtoken := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "addon-operator-prom-token",
+			Namespace: r.addonOperatorNamespace,
+		},
+		Data: map[string][]byte{
+			"token": []byte("mock-token"),
+		},
+	}
+
+	uncachedC.
+		On("Get",
+			testutil.IsContext,
+			mock.Anything,
+			mock.IsType(&corev1.Secret{}), mock.Anything).Run(func(args mock.Arguments) {
+		secretADOtoken.DeepCopyInto(args.Get(2).(*corev1.Secret))
+	}).
+		Return(nil)
+
+	uncachedC.
+		On("Get",
+			testutil.IsContext,
+			mock.Anything,
+			mock.IsType(&corev1.Secret{}), mock.Anything).
 		Return(testutil.NewTestErrNotFound())
+
+	c.
+		On("Create", mock.Anything, mock.IsType(&corev1.Secret{}), mock.Anything).Return(nil).Maybe()
+
+	c.On("Update", mock.Anything, mock.IsType(&corev1.Secret{}), mock.Anything).Return(nil)
+
 	c.On("Create", testutil.IsContext, mock.IsType(&monitoringv1.ServiceMonitor{}), mock.Anything).
 		Run(func(args mock.Arguments) {
+
 			serviceMonitor := args.Get(1).(*monitoringv1.ServiceMonitor)
 			assert.Equal(t, "https", serviceMonitor.Spec.Endpoints[0].Port)
-			assert.Equal(t, "/var/run/secrets/kubernetes.io/serviceaccount/token", serviceMonitor.Spec.Endpoints[0].BearerTokenFile)
```

Comment:
Can we add this assertion back ? (We assert `Authorization` attribute is present in the endpoint slice instead)